sn_node/src/node/bootstrap/join.rs

```diff
@@ -443,19 +469,42 @@ impl<'a> Joiner<'a> {
 
     #[tracing::instrument(skip(self))]
     async fn receive_join_response(&mut self) -> Result<(JoinResponse, Peer)> {
-        while let Some(MsgFromPeer { sender, wire_msg }) = self.incoming_msgs.recv().await {
-            // We are interested only in `JoinResponse` type of messages
+        loop {
+            let (msg, sender) = self.receive_node_msg().await?;
+            match msg {
+                NodeMsg::JoinResponse(resp) => return Ok((*resp, sender)),
+                _ => {
+                    trace!("Bootstrap message discarded: sender: {sender:?} msg: {msg:?}")
```

Comment:
I wonder if this log is clearer as 
```suggestion
                    trace!("Non-Bootstrap message received and discarded: sender: {sender:?} msg: {msg:?}")
```